src/mmds/src/data_store.rs

```diff
@@ -66,7 +76,12 @@ impl Mmds {
 
     pub fn patch_data(&mut self, patch_data: Value) -> Result<(), Error> {
         self.check_data_store_initialized()?;
-        super::json_patch(&mut self.data_store, &patch_data);
+        let mut data_store_clone = self.data_store.clone();
+        super::json_patch(&mut data_store_clone, &patch_data);
+        if data_store_clone.to_string().len() > self.data_store_limit {
+            return Err(Error::DataStoreLimitExceeded);
+        }
+        self.data_store = data_store_clone;
```

Comment:
I think we can avoid the `to_string` here by checking the current length of the data store. As it is a `Value`, we can do `data_store.as_str().unwrap().as_bytes().len()` (unsure if we should unwrap or do error handling). It avoids extra allocation and deallocation of the `String` object.

Reply:
Your observation is a valid one but according to `Value` documentation as_str will return some("a string") only when Value is a String, otherwise the result will be None.

In our use case this does not really makes sense because we usually store maps.